build/dev-env.sh

```diff
@@ -46,22 +46,22 @@ if ! command -v helm &> /dev/null; then
 fi
 
 HELM_VERSION=$(helm version 2>&1 | grep -oE 'v[0-9]+\.[0-9]+\.[0-9]+') || true
-if [[ ${HELM_VERSION} < "v3.0.0" ]]; then
-  echo "Please upgrade helm to v3.0.0 or higher"
+if [[ ${HELM_VERSION} < "v3.8.0" ]]; then
```

Comment:
Maybe I'm missing something, do we have to be above 3.8?

Reply:
I don't see any code that makes helm v3.8+ a requirement.
I felt sane defaults should be 1-2 releases behind current so I tried this.

Additionally, I have messed up somewhere else in a server check because `make dev-env` is failing with some server check somewhere, because of these changes. Still looking but error message is posted in PR